source/_nav.erb

```diff
@@ -8,17 +8,19 @@
             		<img src='<%= site_url %>/img/logos/dbt-lockup.png' style="height: 48px"/>
 		  </a>
 		</div>
+		<% if defined?(version) %>
 		<div class="navigation-link">
 		     <a class="btn btn-teal btn-shadow github-button" style="background-color:#047377;color:#fff"
-			href="https://github.com/dbt-labs/dbt"
+			href="<%=version._source.url.split("/tree")[0]%>"
 			data-size="large"
 			data-show-count="true"
-			aria-label="Star dbt-labs/dbt on GitHub"
+			aria-label="Star the package on GitHub"
 			target="_blank">
 			     <i data-icon="star"></i>
-			     GitHub
+			     <%= version._source.url.split("/")[3] %>/<%= version._source.url.split("/")[4] %>
 			</a>
 		</div>
+		<% end %>
```

Comment:
This approach looks good to me.

The key thing we'd need to do is handle the case where `version._source.url` doesn't conform to the expected pattern. e.g., handle `nil` values or strings that don't split on `/` like we'd expect.